src/librustc_resolve/lib.rs

```diff
@@ -3912,21 +3916,27 @@ impl<'a> Resolver<'a> {
                 }
             }
             Def::ConstParam(..) => {
-                // A const param is always declared in a signature, which is always followed by
-                // some kind of function rib kind (specifically, ItemRibKind in the case of a
-                // normal function), so we can skip the first rib as it will be guaranteed to
-                // (spuriously) conflict with the const param.
-                for rib in &ribs[1..] {
-                    if let ItemRibKind = rib.kind {
-                        // This was an attempt to use a const parameter outside its scope.
-                        if record_used {
-                            resolve_error(
-                                self,
-                                span,
-                                ResolutionError::GenericParamsFromOuterFunction(def),
-                            );
+                let mut ribs = ribs.iter().peekable();
+                if let Some(Rib { kind: FnItemRibKind, .. }) = ribs.peek() {
+                    // When declaring const parameters inside function signatures, the first rib
+                    // is always a `FnItemRibKind`. In this case, we can skip it, to avoid it
+                    // (spuriously) conflicting with the const param.
```

Comment:
Could you give an example of "spuriously conflicting with the const param"? Is there a test?

Reply:
Any function definition with a const generic parameter would conflict. E.g.
https://github.com/rust-lang/rust/blob/f94f85bebde816d78d8a7e6d8ebb75b2a254593c/src/test/ui/const-generics/const-expression-parameter.rs#L4-L6